core.js

```diff
@@ -56,6 +56,19 @@ function _check(buffer, headers, options) {
 	return true;
 }
 
+async function _checkSequence(sequence, tokenizer, options) {
+	options = {
+		position: 0,
+		...options
+	};
+
+	const buffer = Buffer.alloc(minimumBytes);
+
+	await tokenizer.peekBuffer(buffer, {position: options.position, length: 512, mayBeLess: true});
```

Comment:
You can safely use `tokenizer.readBuffer` (advancing the position) once you detected the is PDF kind.

> What you can look for however is the string <xmp:CreatorTool>Adobe Illustrator 24.0 (Windows)</xmp:CreatorTool> or similar, which is near the start of the file, before all the data!

Do you think it possible to `tokenizer.read...` as long we read  _before all the data_ is met?

Reply:
I assume you're talking about `readToken`, correct? If so, we probably can read the buffer starting at the `1350`s byte, read 512 (for safety) and check if the resulting string includes `Adobe Illustrator`? Is that what you're referring to?

Reply:
I have not looked the format at all, but I was (naively) hoping there is a structured way of reading possible. Similar how the zip or png file is done.

Reply:
Here is a screenshot about how it looks. The items marked with a square are those that can (and will) change; specifically the document title (file name you save it as) and some arbitrary length... This is why the code skips `1350` bytes (which gives plenty leeway even for long file names) and reads `512` bytes which should *hopefully* catch the `CreatorTool`

![image](https://user-images.githubusercontent.com/17960496/73776788-110b3a80-4791-11ea-95eb-63c55f02bafc.png)

Reply:
I my god, [Adobe XMP](https://www.adobe.com/products/xmp.html) using [RDF](https://www.w3.org/RDF/), serialized with [RDF/XML](https://en.wikipedia.org/wiki/RDF/XML) inside PDF. And maybe I did not notice all layers yet.

Reply:
If we can somehow figure out the length, then ignore till the title start, ignore title length, then till `CreatorTool`, we could probably use `readToken`... But, at least in this case... It's easier to read 512 bytes than do that. Plus, this method makes it work with other file types, if needed! Thoughts?

Reply:
If there is a header or field length encoded, it could be worth to try to decode that. Your current solution simple, which is also worth something.